src/Symfony/Component/Translation/PseudoLocalizationTranslator.php

```diff
@@ -0,0 +1,358 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class PseudoLocalizationTranslator implements TranslatorInterface
```

Comment:
I would add a phpdoc to make it clear that this should only be used in development env, and never in production (might sound obvious, but sometimes, I like to state the obvious).